### ISSUE TITLE
System.ArgumentException : Unknown type: tuple when deploying a contract

### ISSUE THREAD

**Issue by devedse** [Nethereum/Nethereum#233]
I just tried adding a function to an already existing contract and deploying this through Nethereum and saw the following exception:

```
System.ArgumentException : Unknown type: tuple
Stack Trace:
   at Nethereum.ABI.ABIType.CreateABIType(String typeName)
   at Nethereum.ABI.ArrayType.InitialiseElementType(String name)
   at Nethereum.ABI.ArrayType..ctor(String name)
   at Nethereum.ABI.DynamicArrayType..ctor(String name)
   at Nethereum.ABI.ArrayType.CreateABIType(String typeName)
   at Nethereum.ABI.ABIType.CreateABIType(String typeName)
   at Nethereum.ABI.Model.Parameter..ctor(String type, String name, Int32 order, String serpentSignature)
   at Nethereum.ABI.JsonDeserialisation.ABIDeserialiser.BuildFunctionParameters(List`1 inputs)
   at Nethereum.ABI.JsonDeserialisation.ABIDeserialiser.BuildFunction(IDictionary`2 function)
   at Nethereum.ABI.JsonDeserialisation.ABIDeserialiser.DeserialiseContract(String abi)
   at Nethereum.Contracts.DeployContractTransactionBuilder.BuildEncodedData(String abi, String contractByteCode, Object[] values)
   at Nethereum.Contracts.DeployContractTransactionBuilder.BuildTransaction(String abi, String contractByteCode, String from, Object[] values)
   at Nethereum.Contracts.DeployContract.EstimateGasAsync(String abi, String contractByteCode, String from, Object[] values)
```

This started happening after I added the following function to my contract:

```
function tokensOfOwner(address _owner) external view returns(token[] ownerTokens) {
   ...
}
```

**Comment by juanfranblanco** [comment]
Tuples, return structure are not implemented yet

**Comment by juanfranblanco** [comment]
As you have noticed ;)

**Comment by devedse** [comment]
Where a tuple is basically a struct in solidity then I assume?

**Comment by stephen-squidink** [comment]
Hey, when do you plan to implement this, would love to see this on the roadmap. Thanks :)

**Comment by qbit86** [comment]
> Tuples, return structure are not implemented yet

😰 Any ETA on this?

**Comment by juanfranblanco** [comment]
None yet, it is a priority but not much time at the moment.

**Comment by alphamarke** [comment]
chance to get it soon?

**Comment by juanfranblanco** [comment]
Released on 3.0

**Comment by devedse** [comment]
Nice, thanks :)
